torch/csrc/jit/codegen/cuda/scheduler/registry.cpp

```diff
@@ -957,6 +1001,14 @@ class PointWiseScheduler : public SchedulerEntry {
       return false;
     }
 
+    if (!scheduler_utils::allMatchingViews(fusion) &&
+        SchedulerTopologyChecker::hasViewNotDependentOnRef(
+            fusion, {getReferenceTensorView(fusion)})) {
```

Comment:
The first check here would pass. Then the scheduler would propagate T2 onto the rest of the fusion, then schedule T1. We definitely could/should modify the logic since we could just select T2, but the end result should be the same in most cases.

I was just taking the dependent ref approach as a freebie on reduction/normalization schedules when it happens. We can definitely improve and we probably should in the future.

Reply:
canSchedule had issues because of: https://github.com/csarofeen/pytorch/pull/1949#issuecomment-1234398821

This won't have a similar impact, though I agree it can definitely be improved since we could select from many reference candidates. This should be safe for now, jut not optimal.

Reply:
As long as we pick the same reference consistently we should be fine. This doesn't matter for this part if the reference changes based on the order of outputs, though we may want to become more tolerant to such changes. That's just hard to do since we use the order of outputs as the starting point for traversal and we'd have to have some consistent canonicalization that is consistent and deterministic even if the order of those outputs change.

Reply:
Added your example.